### PR TITLE
Allow users to connect to existing contracts - closes #46

### LINKED ISSUE #46: Allow interaction with Scilla contracts by address
Currently, you need to deploy a contract to interact with it. It ought to be possible to connect() to a contract that has already been deployed.

### PR DESCRIPTION
**Pull request by rrw-zilliqa**
See summary.

### CODE REVIEW

**Comment by its-saeed** [comment]
Please update the readme too.

File: src/ScillaContractDeployer.ts [hunk 1]
```diff
@@ -76,6 +78,24 @@ function read(f: string) {
   return t;
 }
 
+/// Allows you to change setup parameters. Available params:
+/// gasPrice, gasLimit, attempts, timeout.
+export function updateSetup(args: any) {
+  if (setup === null) {
+    throw new HardhatPluginError("hardhat-scilla-plugin", "Please call the initZilliqa function.");
+  }
+  let newSetup : Setup = {
```
Comment (its-saeed) [inline src/ScillaContractDeployer.ts:87]: `let merged = {...obj1, ...obj2};`

merged is now the union of obj1 and obj2. Properties in obj2 will overwrite those in obj1.

this syntax is nicer and easier to read.

File: src/ScillaContractsInfoUpdater.ts [hunk 2]
```diff
@@ -28,6 +28,7 @@ type ContractMapByPath = Record<ContractPath, ContractInfo>;
 export const updateContractsInfo = async () => {
   let contractsInfo: ContractMapByName = {};
   const files = glob.sync("contracts/**/*(*.scilla|*.scillib)");
+  console.log(`files = ${files}`);
```
Comment (its-saeed) [inline src/ScillaContractsInfoUpdater.ts:31]: Was it not for debugging?

File: src/ScillaContractInteractor.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  if (ScillaContractDeployer.setup === null) {
+    throw new HardhatPluginError("hardhat-scilla-plugin", "Please call the initZilliqa function.");
+  }
+  let setup = ScillaContractDeployer.setup
+  let zilliqa = setup.zilliqa
+  let codeResult = await zilliqa.blockchain.getSmartContractCode(address);
+  if (codeResult !== undefined && codeResult.result !== undefined
+    && codeResult.result.code !== undefined) {
+    let codeText = codeResult.result.code;
+    // Now parse it. Sadly, need a file for this. Even more sadly,m
```
Comment (sal-zilliqa) [inline src/ScillaContractInteractor.ts:40]: unfinished comment

**Review by its-saeed — DISMISSED** [review]

**Review by sal-zilliqa — DISMISSED** [review]

File: src/ScillaContractsInfoUpdater.ts (hunk 2 above)
Comment (rrw-zilliqa) [reply src/ScillaContractsInfoUpdater.ts]: It was! Thanks - fixed.

File: src/ScillaContractInteractor.ts (hunk 3 above)
Comment (rrw-zilliqa) [reply src/ScillaContractInteractor.ts]: Sorry - fixed.

File: src/ScillaContractDeployer.ts (hunk 1 above)
Comment (rrw-zilliqa) [reply src/ScillaContractDeployer.ts]: I've changed it, but it's not much nicer. If you try the direct approach .. 
let obj : any = { fish, soup }
let result = { .. setup, .. obj }

and soup is undefined, you get setup.soup overridden to undefined :-(

**Review by sal-zilliqa — APPROVED** [review]

**Review by sal-zilliqa — APPROVED** [review]

**Pull request merged**